src/Game/zNPCTypeCommon.cpp

```diff
@@ -504,17 +530,25 @@ void xNPCBasic::PostInit()
 // func_800F4864
 #pragma GLOBAL_ASM("asm/Game/zNPCTypeCommon.s", "__as__10xBaseAssetFRC10xBaseAsset")
 
-// func_800F4888
-#pragma GLOBAL_ASM("asm/Game/zNPCTypeCommon.s", "ImmTranOn__7xPsycheFv")
+void xPsyche::ImmTranOn()
+{
+    flg_psyche = flg_psyche | 1;
+}
 
-// func_800F4898
-#pragma GLOBAL_ASM("asm/Game/zNPCTypeCommon.s", "ImmTranOff__7xPsycheFv")
+void xPsyche::ImmTranOff()
+{
+    flg_psyche = flg_psyche & 0xfffffffe;
+}
 
-// func_800F48A8
-#pragma GLOBAL_ASM("asm/Game/zNPCTypeCommon.s", "ImmTranIsOn__7xPsycheFv")
+bool xPsyche::ImmTranIsOn()
+{
+    return flg_psyche & 1;
+}
 
-// func_800F48B4
-#pragma GLOBAL_ASM("asm/Game/zNPCTypeCommon.s", "HasGoal__7xPsycheFi")
+int32 xPsyche::HasGoal(int32 goal)
+{
+    return FindGoal(goal) != NULL;
```

Comment:
`return FindGoal(goal);`

Reply:
that is an invalid line of code, and return (int32)FindGoal(goal); does not ok